fog/types/src/common.rs

```diff
@@ -45,6 +47,30 @@ impl core::fmt::Display for BlockRange {
     }
 }
 
+impl FromStr for BlockRange {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let block_indices: Vec<u64> = s
+            .split(',')
+            .map(|index_str| {
+                index_str
+                    .parse()
+                    .expect("BlockRange index is not a number.")
+            })
+            .collect();
```

Comment:
This shows up in the excellent test coverage below.
There are 2 different ways to fail:

    One panics causing the process to stop
    The other returns back an error string saying what happened, letting the caller decide if panicking is appropriate.

Unless there is good reason it seems like the failure behavior should be the same for the 2 cases.

I'm not sure where these strings are coming from yet, but generally I lean toward returning an Error over panicking if the api allows it.

Perhaps something like this would make them more consistent:
```suggestion
        let block_indices = s
            .split(',')
            .map(|index_str| {
                index_str
                    .parse()
            })
            .collect::<Result<Vec<_>, _>>().map_err(|_| "BlockRange index is not a number.")?;
```

Reply:
The strings are coming from the command-line, so I think returning an Error is fine. Changed it (and the test) accordingly.